### ISSUE TITLE
Add support for contract constructors (CAP-58)

### ISSUE THREAD

**Issue by dmkozh** [stellar/rs-soroban-env#1438]
This is a tracking issue for [CAP-58](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0058.md) implementation. The change is relatively small can will likely contain just a few per-repo components:

- [ ] XDR changes as per CAP
- [ ] Env changes that implement constructor semantics
- [ ] Core XDR bump (should probably be no-op or almost no-op)

**Comment by anupsdf** [comment]
Isn't there calibration / metering needed for the new host function create_contract_with_constructor?

**Comment by dmkozh** [comment]
Nope, this uses the building blocks that already exist. We only need to add new cost types when we're doing new and non-trivial work (typically calling a third-party library, e.g. for crypto).
